python/revng/support/collect.py

```diff
@@ -51,3 +53,19 @@ def collect_libraries(search_prefixes: AnyPaths) -> Tuple[List[str], Set[str]]:
     # not allowed.
     dependencies = set(chain.from_iterable([get_elf_needed(path) for path in to_load]))
     return (to_load, dependencies)
+
+
+def collect_pipelines(search_prefixes: AnyPaths, components: List[str] | None) -> List[str]:
+    pipelines = collect_files(search_prefixes, ["share", "revng", "pipelines"], "*.yml")
+    if components is None:
+        return pipelines
+
+    result: List[str] = []
+    for pipeline in pipelines:
+        with open(pipeline) as f:
+            data = yaml.safe_load(f)
+            assert "Component" in data, "Pipeline file does not specify its component"
+            if data["Component"] in components:
+                result.append(pipeline)
```

Comment:
Python never inspected the pipelines files before this. Unnecessary coupling.

My original idea was to 1) always load all the pipelines, 2) expose for each artifact what was its component and let `revng graphql --all-artifacts` filter based on that.

If you think that's too annoying, at least push handling of `--component` into C++, ignoring pipelines that to not match one of the whitelisted components.

Reply:
Moved the logic on the pipeline side